### PR TITLE
metadata: respond with the metadata only if the request includes the metadata

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Fixes https://github.com/istio/istio/issues/22546

### CODE REVIEW

File: extensions/metadata_exchange/plugin.h [hunk 1]
```diff
@@ -89,6 +89,8 @@ class PluginContext : public Context {
   inline StringView nodeId() { return rootContext()->nodeId(); }
 
   ::Wasm::Common::TrafficDirection direction_;
+  bool metadata_received_{true};
```
Comment (kyessenov) [inline extensions/metadata_exchange/plugin.h]: HTTP spec allows response data before the request data, so we have to opportunistically guess it's received then.

**Comment by bianpengyuan** [comment]
block in case you want to fix the license date.

File: test/envoye2e/http_metadata_exchange/exchange_xds_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
+// Copyright 2019 Istio Authors
```
[the remark this answers is not in the document]
Comment (kyessenov) [reply test/envoye2e/http_metadata_exchange/exchange_xds_test.go]: Thanks, fixed.

**Comment by kyessenov** [comment]
@bianpengyuan The issue is specifically for the ingress. The client setting is needed for the egress I think, and we should fix in a separate issue/PR.

**Comment by bianpengyuan** [comment]
@kyessenov Yeah for sure in different PR. I am just curious is there already a plan about how to fix it.

**Comment by kyessenov** [comment]
@bianpengyuan No, noone complained yet, and it has some implications for the multi-cluster. Egress through istio is not very common AFAICT.

**Pull request merged**